backend/test/routes/client_coding.test.js

```diff
@@ -0,0 +1,37 @@
+const model = require("../../src/models/client_coding.js");
+const { testRoutes, routeTypes } = require("./index.js");
+
+jest.mock("../../src/models/client_coding.js");
+
+testRoutes([
+  {
+    request: { method: "GET", url: "/projects/1/client-coding" },
+    modelFunction: model.findAllByProjectId,
+    capabilities: ["projects_read_all"],
+    type: routeTypes.General,
+  },
+  {
+    request: { method: "GET", url: "/projects/client-coding/1" },
```

Comment:
this one doesn't make sense to me.
why not just have /client-coding/1

Reply:
Because client coding belongs to a project I wanted to keep it under the projects route. Maybe it should be projects/[project id]/client-coding/1 but then we wouldn't actually use the project id from the route since we already have the primary key for the client_coding table and we don't need to do a lookup to the project table. I imagine the ticket for figuring out the routes will take clear this up.